modules/imgur.py

```diff
@@ -18,6 +22,97 @@ def _prefix(self, data):
             text += "%s " % data["account_url"]
         return text
 
+    @utils.hook("received.message.channel",
+                priority=EventManager.PRIORITY_LOW)
+    def channel_message(self, event):
+        if not event["channel"].get_setting("auto-imgur", False):
+            return
+
+        self._imgur(event)
+
+    def _imgur(self, event):
+        msg = event["message"]
+        reply = ""
+        match_gallery = REGEX_GALLERY.match(msg)
+        match_image = REGEX_IMAGE.match(msg)
+
+        if match_gallery:
+            reply = self._parse_gallery(match_gallery.group(1))
+
+        if match_image and not reply:
+            reply = self._parse_image(match_image.group(1))
+
+        if not reply:
+            return
+
+        event.eat()
+
+        self.events.on("send.stdout").call(target=event[
+            "channel"], module_name="Imgur", server=event["server"],
+                                           message=reply)
+        return
+
+    def _parse_gallery(self, hash):
+        api_key = self.bot.config["imgur-api-key"]
+        result = utils.http.request(URL_GALLERY % hash,
+            headers={"Authorization": "Client-ID %s" % api_key},
+            json=True)
+
+        if result and result.data["success"]:
+            data = result.data["data"]
+            text = ""
+
+            nsfw = utils.irc.bold("[NSFW] ") if data["nsfw"] else ""
+            title = utils.irc.bold(data["title"]) + " " if data["title"] else ""
+            views = data["views"]
+            time = datetime.utcfromtimestamp(data["datetime"]). \
+                strftime("%e %b, %Y at %H:%M")
+            ups = utils.irc.color(str(data["ups"]) + "▲", utils.consts.GREEN)
+            downs = utils.irc.color("▼" + str(data["downs"]), utils.consts.RED)
+            images = data["images_count"]
+            image_plural = "" if images is 1 else "s"
+
+            bracket_left = "(" if title or nsfw else ""
+            bracket_right = ")" if title or nsfw else ""
+
+            return "%s%s%sA gallery with %s image%s, %s views, posted %s (%s%s)%s" % \
+               (nsfw, title, bracket_left, images, image_plural, views,
+                time, ups, downs,
+                bracket_right)
+
+
+    def _parse_image(self, hash):
+        api_key = self.bot.config["imgur-api-key"]
+        result = utils.http.request(URL_IMAGE % hash,
+            headers={"Authorization": "Client-ID %s" % api_key},
+            json=True)
+
+        if result and result.data["success"]:
+           # Logging.Log.debug("%s", result.data["data"])
+
+            data = result.data["data"]
+            text = ""
+
+            nsfw = utils.irc.bold("[NSFW] ") if data["nsfw"] else ""
+            title = utils.irc.bold(data["title"]) + " " if data["title"] else ""
```

Comment:
If we have a very long title, that's be a lot of bolding. best to leave formatting to a minimum.